src/useQuery.js

```diff
@@ -25,11 +23,21 @@ module.exports = function useQuery(query, opts = {}) {
   }
 
   React.useEffect(() => {
-    queryReq();
+    queryReq(null, updateResult);
   }, [query, JSON.stringify(opts.variables)]);
 
   return {
     ...state,
-    refetch: () => queryReq({ skipCache: true })
+    refetch: () => queryReq({ skipCache: true }),
+    fetchMore: (fetchMoreOpts, updateResultOverride) => {
+      queryReq(
+        {
+          ...allOpts,
+          ...fetchMoreOpts,
+          variables: { ...opts.variables, ...fetchMoreOpts.variables }
+        },
+        updateResultOverride || updateResult
+      );
+    }
```

Comment:
This would become:

```js
fetchMore: (fetchMoreOptions) => queryReq(fetchMoreOptions)
```

We don't need to merge the options here - they are already bound to the `queryReq` by the closure in useClientRequest which does the merging/overriding.

Reply:
In this merge I was also doing a deeper merge on `options.variables` so that only the ones that are changing need to be passed in each time. As discussed, I've updated this to only do the merge on the variables & pass the rest through.